pkg/estimator/server/estimate.go

```diff
@@ -73,8 +73,11 @@ func (es *AccurateSchedulerEstimatorServer) estimateReplicas(
 		tolerations = requirements.NodeClaim.Tolerations
 	}
 
-	// TODO(Garrybest): design a framework and make filter and score plugins
 	var res int32
+	replicas, err := es.estimateFramework.RunEstimateReplicasPlugins(ctx, &requirements)
+	if err != nil {
+		return replicas, err
+	}
 	processNode := func(i int) {
```

Comment:
Or set the snapshot as the input, maybe it's better?

Reply:
This snapshot is different from kube-scheduler. `kube-scheduler` only schedules one pod simultaneously, so snapshot could be a part of handler. However, estimator could handle several requests simultaneously, they do have their own snapshots. As a result, take snapshot as a member field of handler may not be resonable.

Reply:
Got it. Then we can add `cache *schedcache.Cache` in the Handler if needed ?

Reply:
This approach is feasible I think.

But we should know that if each plugin builds the snapshot itself, there is a possibility that the snapshot built by two plugins may be different because the cache is subject to change all the time. I don't see any issues here though.

Reply:
I think we can assume all plugins share one snapshot at the same time. Cluster changes all the time. Event if plugin has their own snapshot based on that time, there still could be some changes during caculation. Normally, estimation will finish in 100ms so it may not bring too many differences.